src/main/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -234,7 +234,9 @@ protected int doWork() {
         final SAMFileHeader.SortOrder sortOrder = header.getSortOrder();
 
         final SAMFileHeader outputHeader = header.clone();
-
+        if(ASSUME_SORTED || ASSUME_SORT_ORDER != null) {
+            outputHeader.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+        }
```

Comment:
On your first note, the problem is that if the input is not query-sorted, but rather it is query-grouped, you need to say "ASSUME_SORT_ORDER=queryname" but the output will be the same order as the input...and the writer will complain that the order is incorrect....